### DISCUSSION TITLE
Typescript with createContext, v4

### DISCUSSION THREAD

**Discussion by zsolt-dev** [pmndrs/zustand#1362] [Q&A]
Hi,

trying to migrate to TS. I have googled all the TS examples, read all the TS discussions etc. I have to use createContext() for the SSR.

Here is part of my code:

```

interface StoreInterface {
...
}

const { Provider, useStore } = createContext();

// How to pass the storeInterface here ?
export default useStore;
```

Thank you

**Comment by dai-shi** [comment]
If you are using v4, my recommendation is to use new API `createStore` and `useStore` instead of `zustand/context`.
Please see #1180 for migration guide.
(we probably want to have something like that in docs. cc @dbritto-dev )

**Reply by dai-shi** [reply]
Oops, we already have it. #1291 

https://github.com/pmndrs/zustand/blob/cf0bef6d5efc14606fd7780ac916b3a701715e33/docs/guides/initialize-state-with-props.md (hmm, I wasn't able to find it with this title...)

**Reply by dai-shi** [reply]
#1276 is slightly related.

**Reply by dbritto-dev** [reply]
@dai-shi I can update the doc to add some codesandbox demos and maybe add some examples as well

**Reply by dai-shi** [reply]
that sounds great.
